sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/ViewDescriptorFactory.scala

```diff
@@ -0,0 +1,208 @@
+      if (annotatedMethods.size > 1)
+        throw new IllegalArgumentException("Views can have only one method annotated with @Query")
+
+      val annotatedMethod: SpringRestServiceMethod = annotatedMethods.head
+
+      val returnType = annotatedMethod.javaMethod.getReturnType
+      val returnTypeDescriptor =
+        if (returnType == tableType) tableTypeDescriptor
+        else ProtoMessageDescriptors.generateMessageDescriptors(returnType)
+
+      val queryStr = annotatedMethod.javaMethod.getAnnotation(classOf[Query]).value()
+
+      val query = kalix.View.Query
+        .newBuilder()
+        .setQuery(queryStr)
+        .build()
+
+      val jsonSchema = {
+        val builder = kalix.JsonSchema
+          .newBuilder()
+          .setOutput(returnTypeDescriptor.mainMessageDescriptor.getName)
+
+        if (annotatedMethod.params.exists(_.isInstanceOf[BodyParameter])) {
+          // only define a json body if there is a request body mapping
+          builder
+            .setInput(annotatedMethod.requestProtoMessageName)
+            .setJsonBodyInputField("json_body")
+        }
+
+        builder.build()
+      }
+
+      val view = kalix.View
+        .newBuilder()
+        .setJsonSchema(jsonSchema)
+        .setQuery(query)
+        .build()
+
+      val builder = kalix.MethodOptions.newBuilder()
+      builder.setView(view)
+      val methodOptions = builder.build()
+
+      // since it is a query, we don't actually ever want to handle any request in the SDK
+      // the proxy does the work for us, mark the method as non-callable
+      (KalixMethod(annotatedMethod.copy(callable = false)).withKalixOptions(methodOptions), returnTypeDescriptor)
+    }
+
+    val kalixMethods: Seq[KalixMethod] = queryMethod +: updateMethods
+    val serviceName = nameGenerator.getName(component.getSimpleName)
+    val additionalMessages =
+      Seq(tableTypeDescriptor) ++ (if (tableTypeDescriptor != queryResultDescriptor) Seq(queryResultDescriptor)
+                                   else Nil)
+    ComponentDescriptor(nameGenerator, serviceName, component.getPackageName, kalixMethods, additionalMessages)
+  }
+
+  private def addTableOptionsToUpdateMethod(
+      tableName: String,
+      tableProtoMessage: String,
+      builder: MethodOptions.Builder,
+      transform: Boolean) = {
+    val update = kalix.View.Update
+      .newBuilder()
+      .setTable(tableName)
+      .setTransformUpdates(transform)
+
+    val jsonSchema = kalix.JsonSchema
+      .newBuilder()
+      .setOutput(tableProtoMessage)
+      .build()
+
+    val view = kalix.View
+      .newBuilder()
+      .setUpdate(update)
+      .setJsonSchema(jsonSchema)
+      .build()
+    builder.setView(view)
+  }
+
+  private def valueEntityStateClassOf(valueEntityClass: Class[_]): Class[_] = {
+    valueEntityClass.getGenericSuperclass
+      .asInstanceOf[ParameterizedType]
+      .getActualTypeArguments
+      .head
+      .asInstanceOf[Class[_]]
+  }
```

Comment:
We can generify this one as 'getFirstTypeParam' or something like that. 
We are doing the same for the View type param